src/Request/BasicResolver.php

```diff
@@ -16,7 +16,7 @@ public function resolve()
                 empty($_SESSION) ? [] : $_SESSION,
                 empty($_COOKIE) ? [] : $_COOKIE,
                 static::getRequestHeaders($_SERVER),
-                static::getInputParams($_SERVER, $_POST));
+                static::getInputParams($_SERVER));
```

Comment:
Checking the request method should instead happen here, and pass the correct variable array (get vs post variables) into `getInputParams`. It reduces the amount of "magic" going on throughout.

Reply:
In the current implementation if `$_GET` is empty in a `GET` request the method will attempt to pull body parameters from `php://input`, something that it shouldn't do for `GET` requests. 
I'd suggest splitting `getInputParams` into multiple methods to handle `GET` and other request methods separately, or add an argument that determines whether the `php://input` source should be read or not.

Reply:
Correct, checking whether to read from `php://input` should be done by request type, either in a single place or via a flag. I'm not suggesting that input should be read for GET requests.